### PR TITLE
Add hardware template

### PR DESCRIPTION
**Pull request by GDay**
This adds the option to assign hardware to new hires. A user can be assigned to automatically create an admin to do item for them to complete when this gets added to the new hire.

### CODE REVIEW

File: back/admin/sequences/views.py [hunk 1]
```diff
@@ -375,7 +375,11 @@ def post(self, request, pk, type, template_pk, *args, **kwargs):
         return render(
             request,
             "_sequence_condition.html",
-            {"condition": condition, "object": condition.sequence},
+            {
+                "condition": condition,
+                "object": condition.sequence,
+                "sequence": condition.sequence,
```
Comment (GDay) [inline back/admin/sequences/views.py:381]: Unrelated fix.

File: back/admin/hardware/forms.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        )
+        self.helper.layout = layout
+
+    class Meta:
+        model = Hardware
+        exclude = ("template",)
+
+    def clean(self):
+        cleaned_data = super().clean()
+        assigned_to = cleaned_data.get("assigned_to", None)
```
Comment (cscheng) [inline back/admin/hardware/forms.py:51]: `None` argument is redundant here, as it's already the default.

File: back/admin/hardware/models.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            notification_type=self.notification_add_type
+            if add
+            else self.notification_remove_type,
+            extra_text=self.name,
+            created_for=user,
+            item_id=self.id,
+        )
+
+    def execute(self, user):
+        from admin.admin_tasks.models import AdminTask
```
Comment (cscheng) [inline back/admin/hardware/models.py:53]: I don't think there is a circular import here, as `admin_tasks.models` doesn't import from `hardware.models`, so this import can be at the top.

File: back/admin/hardware/tests.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    assert Notification.objects.count() == 4
+
+    # create admin task based on buddy
+    admin2 = admin_factory()
+    emp3 = employee_factory(buddy=admin2)
+    hardware.person_type = Hardware.PersonType.BUDDY
+    hardware.save()
+
+    hardware.execute(emp3)
+    assert AdminTask.objects.filter(
```
Comment (cscheng) [inline back/admin/hardware/tests.py:84]: May be assert hardware and notification count here as well to make it consistents with the assertions of earlier executions?

File: back/admin/people/templates/new_hire_detail.html [hunk 5]
```diff
@@ -150,5 +150,24 @@ <h3 class="card-title">{% translate "Latest activity..." %}</h3>
         </div>
       </div>
     </div>
+    {% if object.hardware.all %}
```
Comment (cscheng) [inline back/admin/people/templates/new_hire_detail.html:153]: Why not a separate subpage as with the Tasks etc.?

**Review by cscheng — COMMENTED** [review]
In the SequenceView, hardware templates/items cannot be dragged to before/after conditions. Is this intentional?

File: back/admin/people/templates/new_hire_detail.html (hunk 5 above)
Comment (GDay) [reply back/admin/people/templates/new_hire_detail.html:153]: It's too basic right now to create a new page for it. This is sufficient for now, maybe it will get migrated to a separate page when more features get added to this.

**Comment by GDay** [comment]
> In the SequenceView, hardware templates/items cannot be dragged to before/after conditions. Is this intentional?

Good catch! This happened due to a rebase with some related changes. Thanks.
